core/src/test/scala/unit/kafka/server/ClientQuotasRequestTest.scala

```diff
@@ -223,8 +279,34 @@ class ClientQuotasRequestTest extends BaseRequestTest {
     alterEntityQuotas(entity, Map((ProducerByteRateProp -> Some(10000.5))), validateOnly = true)
   }
 
+  @Test
+  def testAlterClientQuotasInvalidEntityCombination(): Unit = {
```

Comment:
For these three tests, would it make sense to also verify that the error message? `InvalidRequestException` are thrown all over the place so we may get one but not for the expected cause. Your last commit suggests that you got caught by this already, is it? https://github.com/apache/kafka/pull/9628/commits/1b0ec2a3ca79dc8da5908c41afc4ef87d2288d84#diff-f02b619b5cd14e83f7e21dbe211b3f88336f825e1ee5c630fee32b8a0fbe3d20R294